samples/widgets/js-api-widget/README.md

```diff
@@ -1,77 +1,70 @@
+
+```javascript
+import * as Legend from "esri/widgets/Legend";
+import * as LegendVM from "esri/widgets/Legend/LegendViewModel";
 ```
 
-Create an interface `State` to define the specifications for the `legendWidgetVM` and `layerInfo` properties.
+Create an interface `State` to define the specifications for the `layerInfo` property.
+
 ```javascript
 interface State{
-  legendWidgetVM: LegendVM;
   layerInfo: any
 }
-
 ```
 
-Extend the `React.PureComponent` class and use the `AllWidgetProps` type and implement the `State` interface. The `legendWidget` property is used for the `Legend` widget so it can be referenced throughout the widget. 
+Extend the `React.PureComponent` class and use the `AllWidgetProps` type and implement the `State` interface. The `legendWidget` property is used for the `Legend` widget so it can be referenced throughout the widget.
+
 ```javascript
 export default class Widget extends React.PureComponent<AllWidgetProps<{}>, State>{
   apiWidgetContainer: React.RefObject<HTMLDivElement>;
   legendWidget: Legend;
 
-  mapView: __esri.MapView | __esri.SceneView;
-
-  watchHandle: __esri.Handle;
-
-
 // the class constructor assigns the initial this.state and this.apiWidgetContainer to get references to the DOM nodes.
   constructor(props){
     super(props);
-    this.state = {legendWidgetVM: null, layerInfo: null}
+    this.state = { layerInfo: null }
     this.apiWidgetContainer = React.createRef();
   }
-
 }
 ```
 
-The `createAPIWidget` method basically creates the `Legend` widget and passes the properties into the `new Legend` constructor. It also creates the `Legend` view model and updates the UI by calling `setState`.
-`this.watchHandle` watches the `activeLayerInfos.length`, property and uses `setState` to update the `layerInfo` and assigns at index 0. This will provide the title of the layer. 
-  ```javascript
-  createAPIWidget(){
-    if(!this.mapView){
-      return;
-    }
-    if(!this.legendWidget && this.apiWidgetContainer.current){
-      this.legendWidget = new Legend({
-        view: this.mapView,
-        container: this.apiWidgetContainer.current
-      })
-    }
-    
-    if(!this.state.legendWidgetVM){
-      const vm = new LegendVM({
-        view: this.mapView,
-      });
-      this.setState({
-        legendWidgetVM: vm
-      })
-
-      this.watchHandle = vm.watch('activeLayerInfos.length', () => {
-        this.setState({
-          layerInfo: vm.activeLayerInfos.getItemAt(0)
-        })
-      })
-    }
+The `onActiveViewChange` method gets called when the map view changes, and creates the `Legend` widget and passes the properties into the `new Legend` constructor. It also creates the `Legend` view model and updates the UI by calling `setState`.
+
+```javascript
+onActiveViewChange = (jimuMapView: JimuMapView) => {
+  if (!(jimuMapView && jimuMapView.view)) {
+    return;
```

Comment:
on active view change, do you need to update the view of the `Legend`?

Reply:
Good point - thanks. I just put in an update to fix this.